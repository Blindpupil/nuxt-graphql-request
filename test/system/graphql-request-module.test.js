const config = require('../fixture/nuxt.config.js');
const request = require('request-promise-native');
const { Nuxt, Builder } = require('nuxt');

const url = (path) => `http://localhost:3000${path}`;
const get = (path) => request(url(path));

let nuxt;
let addTemplate;

const setupNuxt = async (config) => {
  nuxt = new Nuxt(config);

  // Spy addTemplate
  addTemplate = nuxt.moduleContainer.addTemplate = jest.fn(nuxt.moduleContainer.addTemplate);

  const build = new Builder(nuxt);

  await build.validatePages();
  await build.generateRoutesAndFiles();
  await nuxt.listen(3000);
};

jest.setTimeout(10000);

describe('Nuxt GraphQL Request', () => {
  beforeAll(async () => {
    nuxt = new Nuxt(config);

    // Spy addTemplate
    addTemplate = nuxt.moduleContainer.addTemplate = jest.fn(nuxt.moduleContainer.addTemplate);

    await new Builder(nuxt).build();
    await nuxt.listen(3000);
  }, 60000);

  afterAll(async () => {
    await nuxt.close();
  });

  test('SSR', async () => {
    const html = await get('/');
    expect(html).toContain('Belgium');
  });

  test('CSR', async () => {
    const window = await nuxt.renderAndGetWindow(url('/'));

    window.onNuxtReady(() => {
      const html = window.document.body.innerHTML;
      expect(html).toContain('Belgium');
    });
  });
});

describe('With AST', () => {
  beforeAll(async () => {
    await setupNuxt(config);
  });

  afterAll(async () => {
    await nuxt.close();
  });

  test('SSR', async () => {
    const html = await get('/with-ast');
    expect(html).toContain('Belgium');
  });

  test('CSR', async () => {
    const window = await nuxt.renderAndGetWindow(url('/with-ast'));

    window.onNuxtReady(() => {
      const html = window.document.body.innerHTML;
      expect(html).toContain('Belgium');
    });
  });
});

describe('With .gql Import', () => {
  beforeAll(async () => {
    await setupNuxt(config);
  });

  afterAll(async () => {
    await nuxt.close();
  });

  test('SSR', async () => {
    const html = await get('/import');
    expect(html).toContain('Belgium');
  });

  test('CSR', async () => {
    const window = await nuxt.renderAndGetWindow(url('/import'));

    window.onNuxtReady(() => {
      const html = window.document.body.innerHTML;
      expect(html).toContain('Belgium');
    });
  });
});

describe('Uses runtime config', () => {
    beforeAll(async () => {
        const newConfig = {
            ...config,
            publicRuntimeConfig: {
                GRAPHQL_ENDPOINT: config.endpoint,
            },
        };

        delete newConfig.endpoint;

        await setupNuxt(newConfig);
    });

    afterAll(async () => {
        await nuxt.close();
    });

    test('SSR', async () => {
        const html = await get('/');
        expect(html).toContain('Belgium');
    });

    test('CSR', async () => {
        const window = await nuxt.renderAndGetWindow(url('/'));

        window.onNuxtReady(() => {
            const html = window.document.body.innerHTML;
            expect(html).toContain('Belgium');
        });
    });
});
