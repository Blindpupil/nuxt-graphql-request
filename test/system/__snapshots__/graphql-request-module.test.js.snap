// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nuxt GraphQL Request SSR 1`] = `
"<body >
    <div data-server-rendered=\\"true\\" id=\\"__nuxt\\"><!----><div id=\\"__layout\\"><div><h2>Countries</h2> <ul><li>
      Andorra
    </li><li>
      Austria
    </li><li>
      Åland
    </li><li>
      Belgium
    </li><li>
      Saint Barthélemy
    </li><li>
      Cyprus
    </li><li>
      Germany
    </li><li>
      Estonia
    </li><li>
      Spain
    </li><li>
      Finland
    </li><li>
      France
    </li><li>
      French Guiana
    </li><li>
      Guadeloupe
    </li><li>
      Greece
    </li><li>
      Ireland
    </li><li>
      Italy
    </li><li>
      Lithuania
    </li><li>
      Luxembourg
    </li><li>
      Latvia
    </li><li>
      Monaco
    </li><li>
      Montenegro
    </li><li>
      Saint Martin
    </li><li>
      Martinique
    </li><li>
      Malta
    </li><li>
      Netherlands
    </li><li>
      Saint Pierre and Miquelon
    </li><li>
      Portugal
    </li><li>
      Réunion
    </li><li>
      Slovenia
    </li><li>
      Slovakia
    </li><li>
      San Marino
    </li><li>
      French Southern Territories
    </li><li>
      Vatican City
    </li><li>
      Kosovo
    </li><li>
      Mayotte
    </li></ul> <h2>Star wars movies</h2> <ul><li>
      A New Hope
    </li><li>
      The Empire Strikes Back
    </li><li>
      Return of the Jedi
    </li><li>
      The Phantom Menace
    </li><li>
      Attack of the Clones
    </li><li>
      Revenge of the Sith
    </li></ul></div></div></div><script>window.__NUXT__=(function(a,b){return {layout:\\"default\\",data:[{countries:[{name:\\"Andorra\\"},{name:\\"Austria\\"},{name:\\"Åland\\"},{name:\\"Belgium\\"},{name:\\"Saint Barthélemy\\"},{name:\\"Cyprus\\"},{name:\\"Germany\\"},{name:\\"Estonia\\"},{name:\\"Spain\\"},{name:\\"Finland\\"},{name:\\"France\\"},{name:\\"French Guiana\\"},{name:\\"Guadeloupe\\"},{name:\\"Greece\\"},{name:\\"Ireland\\"},{name:\\"Italy\\"},{name:\\"Lithuania\\"},{name:\\"Luxembourg\\"},{name:\\"Latvia\\"},{name:\\"Monaco\\"},{name:\\"Montenegro\\"},{name:\\"Saint Martin\\"},{name:\\"Martinique\\"},{name:\\"Malta\\"},{name:\\"Netherlands\\"},{name:\\"Saint Pierre and Miquelon\\"},{name:\\"Portugal\\"},{name:\\"Réunion\\"},{name:\\"Slovenia\\"},{name:\\"Slovakia\\"},{name:\\"San Marino\\"},{name:\\"French Southern Territories\\"},{name:\\"Vatican City\\"},{name:\\"Kosovo\\"},{name:\\"Mayotte\\"}],movies:[{title:\\"A New Hope\\"},{title:\\"The Empire Strikes Back\\"},{title:\\"Return of the Jedi\\"},{title:\\"The Phantom Menace\\"},{title:\\"Attack of the Clones\\"},{title:\\"Revenge of the Sith\\"}]}],fetch:{},error:a,serverRendered:true,routePath:b,config:{_app:{basePath:b,assetsPath:\\"\\\\u002F_nuxt\\\\u002F\\",cdnURL:a}}}}(null,\\"\\\\u002F\\"));</script><script src=\\"/_nuxt/e65d139.js\\" defer></script><script src=\\"/_nuxt/87dc927.js\\" defer></script><script src=\\"/_nuxt/01a9223.js\\" defer></script><script src=\\"/_nuxt/4ed9f77.js\\" defer></script><script src=\\"/_nuxt/019d44c.js\\" defer></script>
  </body>"
`;

exports[`Uses runtime config SSR 1`] = `
"<body >
    <div data-server-rendered=\\"true\\" id=\\"__nuxt\\"><!----><div id=\\"__layout\\"><div><h2>Countries</h2> <ul><li>
      Andorra
    </li><li>
      Austria
    </li><li>
      Åland
    </li><li>
      Belgium
    </li><li>
      Saint Barthélemy
    </li><li>
      Cyprus
    </li><li>
      Germany
    </li><li>
      Estonia
    </li><li>
      Spain
    </li><li>
      Finland
    </li><li>
      France
    </li><li>
      French Guiana
    </li><li>
      Guadeloupe
    </li><li>
      Greece
    </li><li>
      Ireland
    </li><li>
      Italy
    </li><li>
      Lithuania
    </li><li>
      Luxembourg
    </li><li>
      Latvia
    </li><li>
      Monaco
    </li><li>
      Montenegro
    </li><li>
      Saint Martin
    </li><li>
      Martinique
    </li><li>
      Malta
    </li><li>
      Netherlands
    </li><li>
      Saint Pierre and Miquelon
    </li><li>
      Portugal
    </li><li>
      Réunion
    </li><li>
      Slovenia
    </li><li>
      Slovakia
    </li><li>
      San Marino
    </li><li>
      French Southern Territories
    </li><li>
      Vatican City
    </li><li>
      Kosovo
    </li><li>
      Mayotte
    </li></ul> <h2>Star wars movies</h2> <ul><li>
      A New Hope
    </li><li>
      The Empire Strikes Back
    </li><li>
      Return of the Jedi
    </li><li>
      The Phantom Menace
    </li><li>
      Attack of the Clones
    </li><li>
      Revenge of the Sith
    </li></ul></div></div></div><script>window.__NUXT__=(function(a,b){return {layout:\\"default\\",data:[{countries:[{name:\\"Andorra\\"},{name:\\"Austria\\"},{name:\\"Åland\\"},{name:\\"Belgium\\"},{name:\\"Saint Barthélemy\\"},{name:\\"Cyprus\\"},{name:\\"Germany\\"},{name:\\"Estonia\\"},{name:\\"Spain\\"},{name:\\"Finland\\"},{name:\\"France\\"},{name:\\"French Guiana\\"},{name:\\"Guadeloupe\\"},{name:\\"Greece\\"},{name:\\"Ireland\\"},{name:\\"Italy\\"},{name:\\"Lithuania\\"},{name:\\"Luxembourg\\"},{name:\\"Latvia\\"},{name:\\"Monaco\\"},{name:\\"Montenegro\\"},{name:\\"Saint Martin\\"},{name:\\"Martinique\\"},{name:\\"Malta\\"},{name:\\"Netherlands\\"},{name:\\"Saint Pierre and Miquelon\\"},{name:\\"Portugal\\"},{name:\\"Réunion\\"},{name:\\"Slovenia\\"},{name:\\"Slovakia\\"},{name:\\"San Marino\\"},{name:\\"French Southern Territories\\"},{name:\\"Vatican City\\"},{name:\\"Kosovo\\"},{name:\\"Mayotte\\"}],movies:[{title:\\"A New Hope\\"},{title:\\"The Empire Strikes Back\\"},{title:\\"Return of the Jedi\\"},{title:\\"The Phantom Menace\\"},{title:\\"Attack of the Clones\\"},{title:\\"Revenge of the Sith\\"}]}],fetch:{},error:a,serverRendered:true,routePath:b,config:{graphql:{clients:{countries:{endpoint:\\"https:\\\\u002F\\\\u002Fcountries.trevorblades.com\\\\u002F\\"},starWars:{endpoint:\\"https:\\\\u002F\\\\u002Fswapi-graphql.netlify.app\\\\u002F.netlify\\\\u002Ffunctions\\\\u002Findex\\"}}},_app:{basePath:b,assetsPath:\\"\\\\u002F_nuxt\\\\u002F\\",cdnURL:a}}}}(null,\\"\\\\u002F\\"));</script><script src=\\"/_nuxt/e65d139.js\\" defer></script><script src=\\"/_nuxt/87dc927.js\\" defer></script><script src=\\"/_nuxt/01a9223.js\\" defer></script><script src=\\"/_nuxt/4ed9f77.js\\" defer></script><script src=\\"/_nuxt/019d44c.js\\" defer></script>
  </body>"
`;

exports[`Using batch queries SSR 1`] = `
"<body >
    <div data-server-rendered=\\"true\\" id=\\"__nuxt\\"><!----><div id=\\"__layout\\"><div><h2>Countries using EUR</h2> <ul><li>
      Andorra
    </li><li>
      Austria
    </li><li>
      Åland
    </li><li>
      Belgium
    </li><li>
      Saint Barthélemy
    </li><li>
      Cyprus
    </li><li>
      Germany
    </li><li>
      Estonia
    </li><li>
      Spain
    </li><li>
      Finland
    </li><li>
      France
    </li><li>
      French Guiana
    </li><li>
      Guadeloupe
    </li><li>
      Greece
    </li><li>
      Ireland
    </li><li>
      Italy
    </li><li>
      Lithuania
    </li><li>
      Luxembourg
    </li><li>
      Latvia
    </li><li>
      Monaco
    </li><li>
      Montenegro
    </li><li>
      Saint Martin
    </li><li>
      Martinique
    </li><li>
      Malta
    </li><li>
      Netherlands
    </li><li>
      Saint Pierre and Miquelon
    </li><li>
      Portugal
    </li><li>
      Réunion
    </li><li>
      Slovenia
    </li><li>
      Slovakia
    </li><li>
      San Marino
    </li><li>
      French Southern Territories
    </li><li>
      Vatican City
    </li><li>
      Kosovo
    </li><li>
      Mayotte
    </li></ul> <h2>Countries using AUD</h2> <ul><li>
      Australia
    </li><li>
      Cocos [Keeling] Islands
    </li><li>
      Christmas Island
    </li><li>
      Heard Island and McDonald Islands
    </li><li>
      Kiribati
    </li><li>
      Norfolk Island
    </li><li>
      Nauru
    </li><li>
      Tuvalu
    </li></ul></div></div></div><script>window.__NUXT__=(function(a){return {layout:\\"default\\",data:[{eur:[{name:\\"Andorra\\"},{name:\\"Austria\\"},{name:\\"Åland\\"},{name:\\"Belgium\\"},{name:\\"Saint Barthélemy\\"},{name:\\"Cyprus\\"},{name:\\"Germany\\"},{name:\\"Estonia\\"},{name:\\"Spain\\"},{name:\\"Finland\\"},{name:\\"France\\"},{name:\\"French Guiana\\"},{name:\\"Guadeloupe\\"},{name:\\"Greece\\"},{name:\\"Ireland\\"},{name:\\"Italy\\"},{name:\\"Lithuania\\"},{name:\\"Luxembourg\\"},{name:\\"Latvia\\"},{name:\\"Monaco\\"},{name:\\"Montenegro\\"},{name:\\"Saint Martin\\"},{name:\\"Martinique\\"},{name:\\"Malta\\"},{name:\\"Netherlands\\"},{name:\\"Saint Pierre and Miquelon\\"},{name:\\"Portugal\\"},{name:\\"Réunion\\"},{name:\\"Slovenia\\"},{name:\\"Slovakia\\"},{name:\\"San Marino\\"},{name:\\"French Southern Territories\\"},{name:\\"Vatican City\\"},{name:\\"Kosovo\\"},{name:\\"Mayotte\\"}],aud:[{name:\\"Australia\\"},{name:\\"Cocos [Keeling] Islands\\"},{name:\\"Christmas Island\\"},{name:\\"Heard Island and McDonald Islands\\"},{name:\\"Kiribati\\"},{name:\\"Norfolk Island\\"},{name:\\"Nauru\\"},{name:\\"Tuvalu\\"}]}],fetch:{},error:a,serverRendered:true,routePath:\\"\\\\u002Fbatch\\",config:{_app:{basePath:\\"\\\\u002F\\",assetsPath:\\"\\\\u002F_nuxt\\\\u002F\\",cdnURL:a}}}}(null));</script><script src=\\"/_nuxt/e65d139.js\\" defer></script><script src=\\"/_nuxt/17cf908.js\\" defer></script><script src=\\"/_nuxt/01a9223.js\\" defer></script><script src=\\"/_nuxt/4ed9f77.js\\" defer></script><script src=\\"/_nuxt/019d44c.js\\" defer></script>
  </body>"
`;

exports[`With .gql Import SSR 1`] = `
"<body >
    <div data-server-rendered=\\"true\\" id=\\"__nuxt\\"><!----><div id=\\"__layout\\"><div><h2>Countries</h2> <ul><li>
      Andorra
    </li><li>
      Austria
    </li><li>
      Åland
    </li><li>
      Belgium
    </li><li>
      Saint Barthélemy
    </li><li>
      Cyprus
    </li><li>
      Germany
    </li><li>
      Estonia
    </li><li>
      Spain
    </li><li>
      Finland
    </li><li>
      France
    </li><li>
      French Guiana
    </li><li>
      Guadeloupe
    </li><li>
      Greece
    </li><li>
      Ireland
    </li><li>
      Italy
    </li><li>
      Lithuania
    </li><li>
      Luxembourg
    </li><li>
      Latvia
    </li><li>
      Monaco
    </li><li>
      Montenegro
    </li><li>
      Saint Martin
    </li><li>
      Martinique
    </li><li>
      Malta
    </li><li>
      Netherlands
    </li><li>
      Saint Pierre and Miquelon
    </li><li>
      Portugal
    </li><li>
      Réunion
    </li><li>
      Slovenia
    </li><li>
      Slovakia
    </li><li>
      San Marino
    </li><li>
      French Southern Territories
    </li><li>
      Vatican City
    </li><li>
      Kosovo
    </li><li>
      Mayotte
    </li></ul> <h2>Star wars movies</h2> <ul><li>
      A New Hope
    </li><li>
      The Empire Strikes Back
    </li><li>
      Return of the Jedi
    </li><li>
      The Phantom Menace
    </li><li>
      Attack of the Clones
    </li><li>
      Revenge of the Sith
    </li></ul></div></div></div><script>window.__NUXT__=(function(a,b){return {layout:\\"default\\",data:[{countries:[{code:\\"AD\\",name:\\"Andorra\\",currency:a},{code:\\"AT\\",name:\\"Austria\\",currency:a},{code:\\"AX\\",name:\\"Åland\\",currency:a},{code:\\"BE\\",name:\\"Belgium\\",currency:a},{code:\\"BL\\",name:\\"Saint Barthélemy\\",currency:a},{code:\\"CY\\",name:\\"Cyprus\\",currency:a},{code:\\"DE\\",name:\\"Germany\\",currency:a},{code:\\"EE\\",name:\\"Estonia\\",currency:a},{code:\\"ES\\",name:\\"Spain\\",currency:a},{code:\\"FI\\",name:\\"Finland\\",currency:a},{code:\\"FR\\",name:\\"France\\",currency:a},{code:\\"GF\\",name:\\"French Guiana\\",currency:a},{code:\\"GP\\",name:\\"Guadeloupe\\",currency:a},{code:\\"GR\\",name:\\"Greece\\",currency:a},{code:\\"IE\\",name:\\"Ireland\\",currency:a},{code:\\"IT\\",name:\\"Italy\\",currency:a},{code:\\"LT\\",name:\\"Lithuania\\",currency:a},{code:\\"LU\\",name:\\"Luxembourg\\",currency:a},{code:\\"LV\\",name:\\"Latvia\\",currency:a},{code:\\"MC\\",name:\\"Monaco\\",currency:a},{code:\\"ME\\",name:\\"Montenegro\\",currency:a},{code:\\"MF\\",name:\\"Saint Martin\\",currency:a},{code:\\"MQ\\",name:\\"Martinique\\",currency:a},{code:\\"MT\\",name:\\"Malta\\",currency:a},{code:\\"NL\\",name:\\"Netherlands\\",currency:a},{code:\\"PM\\",name:\\"Saint Pierre and Miquelon\\",currency:a},{code:\\"PT\\",name:\\"Portugal\\",currency:a},{code:\\"RE\\",name:\\"Réunion\\",currency:a},{code:\\"SI\\",name:\\"Slovenia\\",currency:a},{code:\\"SK\\",name:\\"Slovakia\\",currency:a},{code:\\"SM\\",name:\\"San Marino\\",currency:a},{code:\\"TF\\",name:\\"French Southern Territories\\",currency:a},{code:\\"VA\\",name:\\"Vatican City\\",currency:a},{code:\\"XK\\",name:\\"Kosovo\\",currency:a},{code:\\"YT\\",name:\\"Mayotte\\",currency:a}],movies:[{title:\\"A New Hope\\"},{title:\\"The Empire Strikes Back\\"},{title:\\"Return of the Jedi\\"},{title:\\"The Phantom Menace\\"},{title:\\"Attack of the Clones\\"},{title:\\"Revenge of the Sith\\"}]}],fetch:{},error:b,serverRendered:true,routePath:\\"\\\\u002Fimport\\",config:{_app:{basePath:\\"\\\\u002F\\",assetsPath:\\"\\\\u002F_nuxt\\\\u002F\\",cdnURL:b}}}}(\\"EUR\\",null));</script><script src=\\"/_nuxt/e65d139.js\\" defer></script><script src=\\"/_nuxt/29820cb.js\\" defer></script><script src=\\"/_nuxt/01a9223.js\\" defer></script><script src=\\"/_nuxt/4ed9f77.js\\" defer></script><script src=\\"/_nuxt/019d44c.js\\" defer></script>
  </body>"
`;

exports[`With AST SSR 1`] = `
"<body >
    <div data-server-rendered=\\"true\\" id=\\"__nuxt\\"><!----><div id=\\"__layout\\"><div><h2>Countries</h2> <ul><li>
      Andorra
    </li><li>
      Austria
    </li><li>
      Åland
    </li><li>
      Belgium
    </li><li>
      Saint Barthélemy
    </li><li>
      Cyprus
    </li><li>
      Germany
    </li><li>
      Estonia
    </li><li>
      Spain
    </li><li>
      Finland
    </li><li>
      France
    </li><li>
      French Guiana
    </li><li>
      Guadeloupe
    </li><li>
      Greece
    </li><li>
      Ireland
    </li><li>
      Italy
    </li><li>
      Lithuania
    </li><li>
      Luxembourg
    </li><li>
      Latvia
    </li><li>
      Monaco
    </li><li>
      Montenegro
    </li><li>
      Saint Martin
    </li><li>
      Martinique
    </li><li>
      Malta
    </li><li>
      Netherlands
    </li><li>
      Saint Pierre and Miquelon
    </li><li>
      Portugal
    </li><li>
      Réunion
    </li><li>
      Slovenia
    </li><li>
      Slovakia
    </li><li>
      San Marino
    </li><li>
      French Southern Territories
    </li><li>
      Vatican City
    </li><li>
      Kosovo
    </li><li>
      Mayotte
    </li></ul> <h2>Star wars movies</h2> <ul><li>
      A New Hope
    </li><li>
      The Empire Strikes Back
    </li><li>
      Return of the Jedi
    </li><li>
      The Phantom Menace
    </li><li>
      Attack of the Clones
    </li><li>
      Revenge of the Sith
    </li></ul></div></div></div><script>window.__NUXT__=(function(a,b){return {layout:\\"default\\",data:[{countries:[{code:\\"AD\\",name:\\"Andorra\\",currency:a},{code:\\"AT\\",name:\\"Austria\\",currency:a},{code:\\"AX\\",name:\\"Åland\\",currency:a},{code:\\"BE\\",name:\\"Belgium\\",currency:a},{code:\\"BL\\",name:\\"Saint Barthélemy\\",currency:a},{code:\\"CY\\",name:\\"Cyprus\\",currency:a},{code:\\"DE\\",name:\\"Germany\\",currency:a},{code:\\"EE\\",name:\\"Estonia\\",currency:a},{code:\\"ES\\",name:\\"Spain\\",currency:a},{code:\\"FI\\",name:\\"Finland\\",currency:a},{code:\\"FR\\",name:\\"France\\",currency:a},{code:\\"GF\\",name:\\"French Guiana\\",currency:a},{code:\\"GP\\",name:\\"Guadeloupe\\",currency:a},{code:\\"GR\\",name:\\"Greece\\",currency:a},{code:\\"IE\\",name:\\"Ireland\\",currency:a},{code:\\"IT\\",name:\\"Italy\\",currency:a},{code:\\"LT\\",name:\\"Lithuania\\",currency:a},{code:\\"LU\\",name:\\"Luxembourg\\",currency:a},{code:\\"LV\\",name:\\"Latvia\\",currency:a},{code:\\"MC\\",name:\\"Monaco\\",currency:a},{code:\\"ME\\",name:\\"Montenegro\\",currency:a},{code:\\"MF\\",name:\\"Saint Martin\\",currency:a},{code:\\"MQ\\",name:\\"Martinique\\",currency:a},{code:\\"MT\\",name:\\"Malta\\",currency:a},{code:\\"NL\\",name:\\"Netherlands\\",currency:a},{code:\\"PM\\",name:\\"Saint Pierre and Miquelon\\",currency:a},{code:\\"PT\\",name:\\"Portugal\\",currency:a},{code:\\"RE\\",name:\\"Réunion\\",currency:a},{code:\\"SI\\",name:\\"Slovenia\\",currency:a},{code:\\"SK\\",name:\\"Slovakia\\",currency:a},{code:\\"SM\\",name:\\"San Marino\\",currency:a},{code:\\"TF\\",name:\\"French Southern Territories\\",currency:a},{code:\\"VA\\",name:\\"Vatican City\\",currency:a},{code:\\"XK\\",name:\\"Kosovo\\",currency:a},{code:\\"YT\\",name:\\"Mayotte\\",currency:a}],movies:[{title:\\"A New Hope\\"},{title:\\"The Empire Strikes Back\\"},{title:\\"Return of the Jedi\\"},{title:\\"The Phantom Menace\\"},{title:\\"Attack of the Clones\\"},{title:\\"Revenge of the Sith\\"}]}],fetch:{},error:b,serverRendered:true,routePath:\\"\\\\u002Fwith-ast\\",config:{_app:{basePath:\\"\\\\u002F\\",assetsPath:\\"\\\\u002F_nuxt\\\\u002F\\",cdnURL:b}}}}(\\"EUR\\",null));</script><script src=\\"/_nuxt/e65d139.js\\" defer></script><script src=\\"/_nuxt/b01485e.js\\" defer></script><script src=\\"/_nuxt/01a9223.js\\" defer></script><script src=\\"/_nuxt/4ed9f77.js\\" defer></script><script src=\\"/_nuxt/019d44c.js\\" defer></script>
  </body>"
`;
